webapp/src/components/AssetFilters/CreatorsFilter/CreatorsFilter.tsx

```diff
@@ -0,0 +1,187 @@
+import { useCallback, useEffect, useMemo, useState } from 'react'
+import classNames from 'classnames'
+import {
+  Box,
+  Dropdown,
+  Icon,
+  Loader,
+  useTabletAndBelowMediaQuery
+} from 'decentraland-ui'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { Pill } from '../../AssetTopbar/SelectedFilters/Pill/Pill'
+import { Props } from './CreatorsFilter.types'
+import { getCreatorsByAddress } from './utils'
+import './CreatorsFilter.css'
+
+type Creator = {
+  name: string
+  address: string
+}
+
+export const CreatorsFilter = ({
+  creators,
+  fetchedCreators,
+  onChange,
+  isLoading,
+  defaultCollapsed = false,
+  onFetchCreators
+}: Props): JSX.Element => {
+  const [isFetchingNames, setIsFetchingNames] = useState(false)
+  const isMobileOrTablet = useTabletAndBelowMediaQuery()
+  const [searchTerm, setSearchTerm] = useState('')
+  const [selectedCreators, setSelectedCreators] = useState<Creator[]>(
+    creators?.length && fetchedCreators
+      ? fetchedCreators.filter(creator => creators.includes(creator.address))
+      : []
+  )
+
+  useEffect(() => {
+    const fetchData = async (addresses: string[]) => {
+      return await getCreatorsByAddress(addresses)
+    }
+
+    if (creators?.length) {
+      setIsFetchingNames(true)
+      fetchData(creators).then(creators => {
```

Comment:
I might be missing something, but isn't it possible to do:
```suggestion
      getCreatorsByAddress(creators).then(creators => {
```

Reply:
true 🤦 good call